draft-ietf-cdni-ci-triggers-rfc8007bis.xml

```diff
@@ -233,414 +240,279 @@
+	processing (i.e. enters "active" state).
+	dCDN SHOULD NOT apply CI/T "invalidate" and "purge" actions to data acquired after the trigger processing started, 
+	but this may not always be achievable, so uCDN cannot count on that.
+   </t>
 
    <t>
-   Security considerations are discussed further in <xref target="security-considerations"/>.</t>
+	If uCDN wishes to invalidate or purge content and then immediately preposition replacement content at the same URLs, 
+	it SHOULD ensure that dCDN has completed the invalidate/purge before initiating the prepositioning. 
+	Otherwise, there is a risk that the dCDN prepositions the new content, then immediately invalidates or
+	purges it (as a result of the two uCDN requests running in parallel).
+	
+	uCDN MAY use the <xref target="execution-policy">Execution Policy</xref> extension to condition the start of preposition 
+	trigger processing on completion of the earlier invalidate/purge trigger(s).
+   </t>
 
    <t>
-   The diamond configuration may lead to inefficient interactions, but
-   the interactions are otherwise harmless. For example:</t>
-
-   <t><list style="symbols"><t>When uCDN issues an "invalidate" CI/T Command, dCDN will
-      receive that command from multiple directly connected uCDNs. The
-      dCDN may schedule multiple such commands separately, and the last
-      scheduled command may affect content already revalidated following
-      execution of the "invalidate" command that was scheduled first.</t>
-
-   <t>If one of dCDN's directly connected uCDNs loses its rights to
-      distribute content, it may issue a CI/T "purge" command. That
-      purge may affect content dCDN could retain because it's
-      distributed by another directly connected uCDN. But, that content
-      can be reacquired by dCDN from the remaining uCDN.</t>
-
-   <t>When uCDN originating an item of content issues a CI/T purge
-      followed by a pre-position, two directly connected uCDNs will pass
-      those commands to dCDN. That dCDN implementation need not merge
-      those operations or notice the repetition, in which case the purge
-      issued by one uCDN will be completed before the other. The first uCDN
-      to finish its purge may then forward the "preposition" trigger,
-      and content pre-positioned as a result might be affected by the
-      still-running purge issued by the other uCDN. However, dCDN
-      will reacquire that content as needed, or when it's asked to
-      pre-position the content by the second uCDN. dCDN
-      implementation could avoid this interaction by knowing which uCDN
-      it acquired the content from, or it could minimize the
-      consequences by recording the time at which the
-      "invalidate"/"purge" command was received and not applying it to
-      content acquired after that time.</t>
-
-   </list>
+	dCDN implementation can choose whether to apply "invalidate" and "purge" triggers to content acquisition that has 
```

Comment:
Noting ongoing conversation on Slack--as worded, I think making the implementation-dependent would risk bringing old content prior to the purge into the cache.

Reply:
changed to 
"dCDN implementation SHOULD apply "invalidate" and "purge" triggers to content acquisition that is in progress 
when the trigger becomes active, to avoid placing purged or invalidated content into cache upon completion of the content acquisition.", and moved it up in the paragraph.